examples/ocaml/test_harness.ml

```diff
@@ -0,0 +1,255 @@
+open Mvalue
+
+type test_block = { block_name : string; block_data : Mvalue.revenue_code list }
+
+type test_data = { test_name : string; blocks : test_block list }
+
+let load_file (input_file : string) : string list =
+  let ic = open_in input_file in
+  let rec build_list line_list =
+    match input_line ic with
+    | next_line -> build_list (next_line :: line_list)
+    | exception End_of_file ->
+        close_in ic;
+        List.rev line_list
+    | exception e ->
+        close_in_noerr ic;
+        raise e
+  in
+  build_list []
+
+let get_file_in_dir (dir_handle : Unix.dir_handle) : string list =
+  let rec build_list file_list =
+    match Unix.readdir dir_handle with
+    | next_file when next_file = "." || next_file = ".." -> build_list file_list
+    | next_file -> build_list (next_file :: file_list)
+    | exception End_of_file ->
+        Unix.closedir dir_handle;
+        file_list
+    | exception e ->
+        Unix.closedir dir_handle;
+        raise e
+  in
+  build_list []
+
+let compare_rev_code code1 code2 : int = compare code1.alias code2.alias
+
+let line2revenue_code line : Mvalue.revenue_code =
+  match String.split_on_char '/' line with
+  | [ head; tail ] -> { alias = head; value = Float.of_string tail }
+  | [] -> failwith "Empty line !"
+  | _ :: tail -> failwith "Line with multiple separators /."
+
+let parse_codes line_list : Mvalue.revenue_code list =
+  match line_list with [] -> [] | _ -> List.map line2revenue_code line_list
+
+let parse_block line_list : test_block =
+  match line_list with
+  | head :: tail when String.starts_with ~prefix:"#" head ->
+      { block_name = head; block_data = parse_codes tail }
+  | _ -> failwith "Data block has not the expected structure."
+
+let rec split_blocks (line_list : string list)
+    (line_list_list : string list list) : string list list =
+  match line_list with
+  | [] -> List.rev (List.map List.rev line_list_list)
+  | line :: tail ->
+      if String.starts_with ~prefix:"#" line then
```

Comment:
Should be fixed by using the parser from the test_framework in the main mlang codebase, instead of doing the same job badly and buggyly.

Reply:
The backend is now usable with compiler 4.11.2.

I have copied selected parts of Mlang internal parser and test parser in order to obtain a independent parser for DGFiP test format. It is incomplete (as it was in Mlang test), but I think it would be a good idea to extend it.